sktime/classification/distance_based/_shape_dtw.py

```diff
@@ -0,0 +1,477 @@
+# Utilities
+import numpy as np
+import pandas as pd
+import math
+from sktime.utils.validation.series_as_features import check_X, check_X_y
+from sktime.utils.data_container import tabularize
+
+# Tuning
+from sklearn.model_selection import GridSearchCV
+from sklearn.model_selection import KFold
+
+# Transforms
+from sktime.transformers.series_as_features.segment \
+    import SlidingWindowSegmenter
+from sktime.transformers.series_as_features.dictionary_based._paa \
+    import PAA
+from sktime.transformers.series_as_features.dwt import DWTTransformer
+from sktime.transformers.series_as_features.slope import SlopeTransformer
+from sktime.transformers.series_as_features.derivative import DerivativeTransformer
+from sktime.transformers.series_as_features.hog1d import HOG1DTransformer
+
+# Classifiers
+from sktime.classification.base import BaseClassifier
+from sktime.classification.distance_based import KNeighborsTimeSeriesClassifier
+
+
+class ShapeDTW(BaseClassifier):
+
+    def __init__(self, n_neighbors=1, subsequence_length=None,
+                 shape_descriptor_function='raw',
+                 shape_descriptor_functions=['raw', 'derivative'],
+                 metric_params=None):
+        """
```

Comment:
the init doc string is a bit of an exception, it goes under the class definition, adding a sentence describing what it does would also be great, the rest looks great! :)